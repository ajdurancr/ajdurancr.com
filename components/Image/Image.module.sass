.container
  width: 100%
  
  img
    height: auto
    width: 100%