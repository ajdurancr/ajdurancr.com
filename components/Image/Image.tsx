import React from 'react'
import classnames from 'classnames'

import styles from './Image.module.sass'

type ImageProps = {
  className?: string
  src: string
  alt: string
}
const Image: React.FC<ImageProps> = ({ className = '', src, alt }) => (
  <div className={classnames(styles.container, className)}>
    <img src={src} alt={alt} />
  </div>

)

export { Image }
