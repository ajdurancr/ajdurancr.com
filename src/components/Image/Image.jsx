import React from 'react'
import propTypes from 'prop-types'
import classnames from 'classnames'

import styles from './Image.styles.sass'

const Image = ({ className, src, alt }) => (
  <div className={classnames(styles.container, className)}>
    <img src={src} alt={alt} />
  </div>

)

Image.defaultProps = {
  className: '',
}

Image.propTypes = {
  className: propTypes.string,
  src: propTypes.string.isRequired,
  alt: propTypes.string.isRequired,
}

export default Image
