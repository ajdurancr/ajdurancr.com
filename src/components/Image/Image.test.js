import React from 'react'
import { shallow } from 'enzyme'

import Image from './Image'

describe('Image', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Image className="testClassName" src="/test/image.svg" alt="Alternative text" />)

    expect(wrapper).toMatchSnapshot()
  })
})
