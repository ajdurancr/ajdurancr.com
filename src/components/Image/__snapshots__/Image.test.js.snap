// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<div
  className="container testClassName"
>
  <img
    alt="Alternative text"
    src="/test/image.svg"
  />
</div>
`;
